src/dbnode/storage/series/series.go

```diff
@@ -204,24 +205,29 @@ func (s *dbSeries) updateBlocksWithLock(
 
 		// Potentially unwire
 		var unwired, shouldUnwire bool
-		// Makes sure that the block has been flushed, which
-		// prevents us from unwiring blocks that haven't been flushed yet which
-		// would cause data loss.
-		if blockState := blockStates[startNano]; blockState.WarmRetrievable {
-			switch cachePolicy {
-			case CacheNone:
-				shouldUnwire = true
-			case CacheRecentlyRead:
-				sinceLastRead := now.Sub(currBlock.LastReadTime())
-				shouldUnwire = sinceLastRead >= wiredTimeout
-			case CacheLRU:
-				// The tick is responsible for managing the lifecycle of blocks that were not
-				// read from disk (not retrieved), and the WiredList will manage those that were
-				// retrieved from disk.
-				shouldUnwire = !currBlock.WasRetrievedFromDisk()
-			default:
-				s.opts.InstrumentOptions().Logger().Fatal(
-					"unhandled cache policy in series tick", zap.Any("policy", cachePolicy))
+		blockStatesSnapshot, bootstrapped := blockStates.Snapshot()
+		// Only use block state snapshot information to make eviction decisions if the block state
+		// has been properly bootstrapped already.
+		if bootstrapped {
```

Comment:
Wondering why adding this check now is important, it should've always been an issue no?

Reply:
it basically "happened" to work before because an empty blockStateSnapshot will have `WarmRetrievable` `false` and `ColdVersion` `0` so you would never evict anything. I noticed this while I was working on the P.R and thought it was really dangerous since basically you come in and think you're programming against data that has been properly bootstrapped but actually in some cases you're (temporarily) running against a zero value so I wanted to make it more explicit so it doesn't get messed up in the future